### PR TITLE
Cnn

### PR DESCRIPTION
**Pull request by b-sherman**
Initial implementation of the convolutional neural networks module.

### CODE REVIEW

File: docs/source/content/deep_learning/cnn.rst [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+.. contents::
+  :local:
+  :depth: 2
+
+
+********
+Overview
+********
+In the last module, we started our dive into deep learning by talking about
+multi-layer perceptrons. In this module, we will learn about **convolutional
```
Comment (hokie45) [inline docs/source/content/deep_learning/cnn.rst:14]: I would define what a perceptron is. I don't think most people would know what that means. Actually just link it to Eric's portion, when he finishes it.

File: docs/source/content/deep_learning/cnn.rst [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+neural networks in that sequential layers are not necessarily fully connected.
+This means that a subset of the input neurons may only feed into a single
+neuron in the next layer. Another interesting feature of CNNs is their inputs.
+With other neural networks we might use vectors as inputs, but with CNNs we
+are typically working with images and other objects with many dimensions.
+*Figure 1* shows some sample images that are each 6 pixels by 6 pixels. The
+first image is colored and has three channels for red, green, and blue values.
+The second image is black-and-white and only has one channel for gray values
+
+.. figure:: _img/Images.png
```
Comment (hokie45) [inline docs/source/content/deep_learning/cnn.rst:25]: add references to images if you used an outside sources, this goes for all images.

**Review by hokie45 — COMMENTED** [review]
Just make sure to add references to the images provided, if you used an outside source. 
Also add in a reference section at the end, you can look at my PR to see how I did it for the other files.

File: docs/source/content/deep_learning/cnn.rst (hunk 1 above)
Comment (b-sherman) [reply docs/source/content/deep_learning/cnn.rst:14]: Thanks! I will link to the other module when available.

File: docs/source/content/deep_learning/cnn.rst (hunk 2 above)
Comment (b-sherman) [reply docs/source/content/deep_learning/cnn.rst:25]: I made all the images so they should be fine as is. I did add in a reference section like you have in your PR though.

**Review by hokie45 — APPROVED** [review]
Thanks for the changes

**Pull request merged**